### PR TITLE
Fix merging duration influence dst duration

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

FIx https://github.com/istio/istio/issues/53800

### CODE REVIEW

File: pkg/proto/merge/merge_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"testing"
+
+	listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"istio.io/istio/pkg/test/util/assert"
+)
+
+func TestMerge(t *testing.T) {
```
Comment (howardjohn) [inline pkg/proto/merge/merge_test.go:25]: This test passes without the rest of the PR. So its probably not testing quite the right thing?

BTW this issue is widespread, this is why we do this everywhere: https://github.com/istio/istio/blob/f524f6319dab722faaeee13d91fd34ab8732c664/pilot/pkg/networking/core/cluster_builder.go#L737

**Comment by howardjohn** [comment]
I think this is probably a good change, but if we do it we should probably make the test more robust + remove all the existing duration clones?

File: pkg/proto/merge/merge_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/proto/merge/merge_test.go:25]: SGTM, let me remove

Comment (hzxuzhonghu) [reply pkg/proto/merge/merge_test.go:26]: @howardjohn Chnaged the test too, should check the dst.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**